File: cpmnuker.cpp
#include "include/cpmnuker.hpp"

CPMNuker::CPMNuker(const std::string& access_key) {
    this->access_key = access_key;
}

int CPMNuker::account_login(const std::string& email, const std::string& password)
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/account_login"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{
            {"account_email", email},
            {"account_password", password}
        }
    );
    json data = json::parse(response.text);
    if(data["ok"] && data["error"] == 0){
        this->auth_token = data["auth"];
        return 0;
    } else {
        return data["error"];
    }
}

bool CPMNuker::account_register(const std::string& email, const std::string& password)
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/account_register"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{
            {"account_email", email},
            {"account_password", password}
        }
    );
    json data = json::parse(response.text);
    if(data["ok"]){
        this->auth_token = data["auth"];
        return true;
    } else {
        return false;
    }
}

bool CPMNuker::account_delete()
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/account_delete"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{{"account_auth", this->auth_token}}
        );
    json data = json::parse(response.text);
    if(data["ok"]){
        return true;
    } else {
        return false;
    }
}

bool CPMNuker::account_set_data(json content)
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/set_data"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{
            {"account_auth", this->auth_token},
            {"content", content.dump()}
        }
    );
    json data = json::parse(response.text);
    if(data["ok"]){
        return true;
    } else {
        return false;
    }
}

json CPMNuker::account_get_data()
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/get_data"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{{"account_auth", this->auth_token}}
    );
    json data = json::parse(response.text);
    return data["data"];
}

bool CPMNuker::account_set_rank()
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/set_rank"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{{"account_auth", this->auth_token}}
        );
    json data = json::parse(response.text);
    if(data["ok"]){
        return true;
    } else {
        return false;
    }
}

bool CPMNuker::account_save_car(const std::string& car_id)
{
    cpr::Response response = cpr::Post(
        cpr::Url{this->BASE_URL + "/unlock_car"},
        cpr::Parameters{{"key", this->access_key}},
        cpr::Payload{
            {"account_auth", this->auth_token},
            {"car_id", car_id}
        }
    );
    json data = json::parse(response.text);
    if(data["ok"]){
        return true;
    } else {
        return false;
    }
}

json CPMNuker::access_key_data()
{
    cpr::Response response = cpr::Get(cpr::Url{this->BASE_URL + "/get_key_data"}, cpr::Parameters{{"key", this->access_key}});
    json data = json::parse(response.text);
    return data;
}

File: cpmnuker.py
import requests
import json

# Copyright (C) Anasov <me@anasov.ly> - All Rights Reserved
# Unauthorized copying of this file, via any medium is strictly prohibited
# Proprietary and confidential
# Written by Anasov <me@anasov.ly>, 05, May, 2024.

BASE_URL: str = "https://cpmnuker.anasov.ly"

class CPMNuker:

    def __init__(self, access_key) -> None:
        self.auth_token = None
        self.access_key = access_key
        
    def login(self, email, password) -> int:
        payload = { "account_email": email, "account_password": password }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/account_login", params=params, data=payload)
        response_decoded = json.loads(response.text)
        if response_decoded.get("ok"):
            self.auth_token = response_decoded.get("auth")
        return response_decoded.get("error")
    
    def register(self, email, password) -> int:
        payload = { "account_email": email, "account_password": password }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/account_register", params=params, data=payload)
        response_decoded = json.loads(response.text)
        if response_decoded.get("ok"):
            self.auth_token = response_decoded.get("auth")
        return response_decoded.get("error")
    
    def delete(self) -> None:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/account_delete", params=params, data=payload)
        response_decoded = json.loads(response.text)

    def get_player_data(self) -> any:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/get_data", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded
    
    def set_player_rank(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_rank", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def get_key_data(self) -> any:
        params = { "key": self.access_key }
        response = requests.get(f"{BASE_URL}/api/get_key_data", params=params)
        response_decoded = json.loads(response.text)
        return response_decoded
    
    def set_player_money(self, amount) -> bool:
        payload = {
            "account_auth": self.auth_token,
            "amount": amount
        }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_money", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def set_player_coins(self, amount) -> bool:
        payload = {
            "account_auth": self.auth_token,
            "amount": amount
        }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_coins", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def set_player_name(self, name) -> bool:
        payload = { "account_auth": self.auth_token, "name": name }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_name", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def set_player_localid(self, id) -> bool:
        payload = { "account_auth": self.auth_token, "id": id }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_id", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")

    def set_player_plates(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/set_plates", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def delete_player_friends(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/delete_friends", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def unlock_w16(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/unlock_w16", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def unlock_horns(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/unlock_horns", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    
    def disable_engine_damage(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/disable_damage", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")

    def unlimited_fuel(self) -> bool:
        payload = { "account_auth": self.auth_token }
        params = { "key": self.access_key }
        response = requests.post(f"{BASE_URL}/api/unlimited_fuel", params=params, data=payload)
        response_decoded = json.loads(response.text)
        return response_decoded.get("ok")
    